lib/referral.ts

```diff
@@ -0,0 +1,60 @@
+import { AxiosRequestConfig } from "axios";
+
+export enum ReferralStatus {
+  PENDING = "pending",
+  REJECTED = "rejected",
+  ACCEPTED = "accepted",
+  COMPLETE = "complete",
+  CANCELLED = "cancelled"
+}
+
+export interface ReferralRequest {
+  email: string;
+  firstName?: string;
+  lastName?: string;
+  phone?: string;
+  venueName?: string;
+  partnerLocationId?: string;
+  doshiiLocationId?: string;
+  doshiiOrganisationId?: string;
+  address: {
```

Comment:
address should be optional, as we cannot guaranty CBA aways send the address for us during the onboarding process.

Reply:
We should check this schema against the Joi schema defined in the PartnerController (in Partner API repo)